File: java/java-sdk/src/main/java/io/kestra/sdk/api/InstanceApi.java
package io.kestra.sdk.api;

import com.fasterxml.jackson.core.type.TypeReference;

import io.kestra.sdk.internal.ApiClient;
import io.kestra.sdk.internal.ApiException;
import io.kestra.sdk.internal.BaseApi;
import io.kestra.sdk.internal.Configuration;
import io.kestra.sdk.internal.Pair;

import io.kestra.sdk.model.InstanceControllerApiActiveServiceList;
import io.kestra.sdk.model.InstanceControllerApiCreateOrUpdateWorkerGroupRequest;
import io.kestra.sdk.model.InstanceControllerApiPluginArtifactListPluginArtifact;
import io.kestra.sdk.model.InstanceControllerApiPluginArtifactListPluginResolutionResult;
import io.kestra.sdk.model.InstanceControllerApiPluginListRequest;
import io.kestra.sdk.model.InstanceControllerApiPluginVersionDetails;
import io.kestra.sdk.model.InstanceControllerApiPluginVersions;
import io.kestra.sdk.model.InstanceControllerApiWorkerGroup;
import io.kestra.sdk.model.InstanceControllerApiWorkerGroupList;
import io.kestra.sdk.model.PagedResultsInstanceControllerApiPluginArtifact;
import io.kestra.sdk.model.PagedResultsInstanceControllerApiServiceInstance;
import io.kestra.sdk.model.PluginArtifact;
import io.kestra.sdk.model.QueryFilter;
import io.kestra.sdk.model.ServiceInstance;
import io.kestra.sdk.model.WorkerCredentialControllerApiWorkerCredential;
import io.kestra.sdk.model.WorkerCredentialControllerApiWorkerList;

import java.io.File;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Instance-owner endpoints under {@code /api/v1/instance/**}: worker groups, worker
 * queues, versioned plugins, worker credentials, services, maintenance and MCP servers.
 * None of these are tenant-scoped — they operate on the whole Kestra cluster and require
 * the {@code IsInstanceOwner} role.
 * <p>
 * Instance-scoped policies ({@code /api/v1/instance/policies/**}) are intentionally NOT
 * here: they live on {@code PoliciesApi} alongside the tenant- and namespace-scoped
 * variants.
 */
public class InstanceApi extends BaseApi {

    private static final String SVG = "image/svg+xml";
    private static final String TEXT_PLAIN = "text/plain";
    private static final String MULTIPART = "multipart/form-data";

    public InstanceApi() {
        super(Configuration.getDefaultApiClient());
    }

    public InstanceApi(ApiClient apiClient) {
        super(apiClient);
    }

    // ========================================================================
    // Worker groups
    // ========================================================================

    public InstanceControllerApiWorkerGroupList listWorkerGroups(
            @jakarta.annotation.Nullable List<QueryFilter> filters) throws ApiException {
        return invoke("GET",
                apiPath("instance", "worker-groups"),
                null, null, filterParams(filters),
                JSON, null,
                new TypeReference<>() {});
    }

    public InstanceControllerApiWorkerGroup createWorkerGroup(
            @jakarta.annotation.Nonnull InstanceControllerApiCreateOrUpdateWorkerGroupRequest request) throws ApiException {
        return invoke("POST",
                apiPath("instance", "worker-groups"),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public InstanceControllerApiWorkerGroup getWorkerGroup(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        return invoke("GET",
                apiPath("instance", "worker-groups", id),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public InstanceControllerApiWorkerGroup updateWorkerGroup(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nonnull InstanceControllerApiCreateOrUpdateWorkerGroupRequest request) throws ApiException {
        return invoke("PUT",
                apiPath("instance", "worker-groups", id),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public void deleteWorkerGroup(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nullable Boolean force) throws ApiException {
        invoke("DELETE",
                apiPath("instance", "worker-groups", id),
                null, queryParams("force", force), null,
                JSON, null,
                null);
    }

    public Map<String, Object> getWorkerGroupCapacity(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        return invoke("GET",
                apiPath("instance", "worker-groups", id, "capacity"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public Map<String, Object> listWorkerGroupWorkers(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        return invoke("GET",
                apiPath("instance", "worker-groups", id, "workers"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public InstanceControllerApiWorkerGroup addWorkerGroupSubscription(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nonnull Map<String, Object> request) throws ApiException {
        return invoke("POST",
                apiPath("instance", "worker-groups", id, "subscriptions"),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public InstanceControllerApiWorkerGroup updateWorkerGroupSubscription(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nonnull String workerQueueId,
            @jakarta.annotation.Nonnull Map<String, Object> request) throws ApiException {
        return invoke("PATCH",
                apiPath("instance", "worker-groups", id, "subscriptions", workerQueueId),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public InstanceControllerApiWorkerGroup removeWorkerGroupSubscription(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nonnull String workerQueueId) throws ApiException {
        return invoke("DELETE",
                apiPath("instance", "worker-groups", id, "subscriptions", workerQueueId),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public Map<String, Object> generateWorkerGroupToken(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nonnull Map<String, Object> request) throws ApiException {
        return invoke("POST",
                apiPath("instance", "worker-groups", id, "tokens"),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public InstanceControllerApiWorkerGroup revokeWorkerGroupToken(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nonnull String tokenId) throws ApiException {
        return invoke("POST",
                apiPath("instance", "worker-groups", id, "tokens", tokenId, "revoke"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public void deleteWorkerGroupToken(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nonnull String tokenId) throws ApiException {
        invoke("DELETE",
                apiPath("instance", "worker-groups", id, "tokens", tokenId),
                null, null, null,
                JSON, null,
                null);
    }

    // ========================================================================
    // Worker queues
    // ========================================================================

    public Map<String, Object> listWorkerQueues() throws ApiException {
        return invoke("GET",
                apiPath("instance", "worker-queues"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public Map<String, Object> createWorkerQueue(
            @jakarta.annotation.Nonnull Map<String, Object> request) throws ApiException {
        return invoke("POST",
                apiPath("instance", "worker-queues"),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public Map<String, Object> getWorkerQueue(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        return invoke("GET",
                apiPath("instance", "worker-queues", id),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public Map<String, Object> updateWorkerQueue(
            @jakarta.annotation.Nonnull String id,
            @jakarta.annotation.Nonnull Map<String, Object> request) throws ApiException {
        return invoke("PUT",
                apiPath("instance", "worker-queues", id),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public void deleteWorkerQueue(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        invoke("DELETE",
                apiPath("instance", "worker-queues", id),
                null, null, null,
                JSON, null,
                null);
    }

    public Map<String, Object> getWorkerQueueSubscribers(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        return invoke("GET",
                apiPath("instance", "worker-queues", id, "subscribers"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    // ========================================================================
    // Versioned plugins
    // ========================================================================

    public PagedResultsInstanceControllerApiPluginArtifact listVersionedPlugins(
            @jakarta.annotation.Nullable Integer page,
            @jakarta.annotation.Nullable Integer size,
            @jakarta.annotation.Nullable List<String> sort,
            @jakarta.annotation.Nullable List<QueryFilter> filters) throws ApiException {
        List<Pair> collectionParams = new ArrayList<>();
        collectionParams.addAll(csvParams("sort", sort));
        collectionParams.addAll(filterParams(filters));
        return invoke("GET",
                apiPath("instance", "versioned-plugins"),
                null, queryParams("page", page, "size", size), collectionParams,
                JSON, null,
                new TypeReference<>() {});
    }

    public Map<String, Object> listAvailableVersionedPlugins() throws ApiException {
        return invoke("GET",
                apiPath("instance", "versioned-plugins", "available"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public Map<String, Object> listAvailableVersionedPluginsForStorage() throws ApiException {
        return invoke("GET",
                apiPath("instance", "versioned-plugins", "available", "storages"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public Map<String, Object> listAvailableVersionedPluginsForSecretManager() throws ApiException {
        return invoke("GET",
                apiPath("instance", "versioned-plugins", "available", "secrets-managers"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public InstanceControllerApiPluginVersions getVersionedPluginDetails(
            @jakarta.annotation.Nonnull String groupId,
            @jakarta.annotation.Nonnull String artifactId) throws ApiException {
        return invoke("GET",
                apiPath("instance", "versioned-plugins", groupId, artifactId),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public byte[] getVersionedPluginIcon(
            @jakarta.annotation.Nonnull String groupId,
            @jakarta.annotation.Nonnull String artifactId,
            @jakarta.annotation.Nullable String version) throws ApiException {
        return invoke("GET",
                apiPath("instance", "versioned-plugins", groupId, artifactId, "icon.svg"),
                null, queryParams("v", version), null,
                SVG, null,
                new TypeReference<>() {});
    }

    public String getVersionedPluginReleaseNotes(
            @jakarta.annotation.Nonnull String groupId,
            @jakarta.annotation.Nonnull String artifactId,
            @jakarta.annotation.Nonnull String version) throws ApiException {
        return invoke("GET",
                apiPath("instance", "versioned-plugins", groupId, artifactId, "release-notes"),
                null, queryParams("version", version), null,
                TEXT_PLAIN, null,
                new TypeReference<>() {});
    }

    public InstanceControllerApiPluginVersionDetails getVersionedPluginDetailsFromVersion(
            @jakarta.annotation.Nonnull String groupId,
            @jakarta.annotation.Nonnull String artifactId,
            @jakarta.annotation.Nonnull String version) throws ApiException {
        return invoke("GET",
                apiPath("instance", "versioned-plugins", groupId, artifactId, version),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public InstanceControllerApiPluginArtifactListPluginArtifact installVersionedPlugins(
            @jakarta.annotation.Nonnull InstanceControllerApiPluginListRequest request) throws ApiException {
        return invoke("POST",
                apiPath("instance", "versioned-plugins", "install"),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public InstanceControllerApiPluginArtifactListPluginResolutionResult resolveVersionedPlugins(
            @jakarta.annotation.Nonnull InstanceControllerApiPluginListRequest request) throws ApiException {
        return invoke("POST",
                apiPath("instance", "versioned-plugins", "resolve"),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    public PluginArtifact uploadVersionedPlugins(
            @jakarta.annotation.Nonnull File file,
            @jakarta.annotation.Nullable Boolean forceInstallOnExistingVersions) throws ApiException {
        Map<String, Object> formParams = new HashMap<>();
        formParams.put("file", file);
        return invoke("POST",
                apiPath("instance", "versioned-plugins", "upload"),
                null, queryParams("forceInstallOnExistingVersions", forceInstallOnExistingVersions), null,
                JSON, MULTIPART, formParams,
                new TypeReference<>() {});
    }

    public InstanceControllerApiPluginArtifactListPluginArtifact uninstallVersionedPlugins(
            @jakarta.annotation.Nonnull InstanceControllerApiPluginListRequest request) throws ApiException {
        return invoke("DELETE",
                apiPath("instance", "versioned-plugins", "uninstall"),
                request, null, null,
                JSON, JSON,
                new TypeReference<>() {});
    }

    // ========================================================================
    // Worker credentials
    // ========================================================================

    public WorkerCredentialControllerApiWorkerList listWorkerCredentials() throws ApiException {
        return invoke("GET",
                apiPath("instance", "workers", "credentials"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public WorkerCredentialControllerApiWorkerCredential getWorkerCredential(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        return invoke("GET",
                apiPath("instance", "workers", "credentials", id),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public WorkerCredentialControllerApiWorkerCredential revokeWorkerCredential(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        return invoke("POST",
                apiPath("instance", "workers", "credentials", id, "revoke"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    // ========================================================================
    // Services
    // ========================================================================

    public InstanceControllerApiActiveServiceList getActiveServices() throws ApiException {
        return invoke("GET",
                apiPath("instance", "services", "active"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public PagedResultsInstanceControllerApiServiceInstance searchServices(
            @jakarta.annotation.Nullable Integer page,
            @jakarta.annotation.Nullable Integer size,
            @jakarta.annotation.Nullable List<String> sort,
            @jakarta.annotation.Nullable List<QueryFilter> filters) throws ApiException {
        List<Pair> collectionParams = new ArrayList<>();
        collectionParams.addAll(csvParams("sort", sort));
        collectionParams.addAll(filterParams(filters));
        return invoke("GET",
                apiPath("instance", "services", "search"),
                null, queryParams("page", page, "size", size), collectionParams,
                JSON, null,
                new TypeReference<>() {});
    }

    public ServiceInstance getService(
            @jakarta.annotation.Nonnull String id) throws ApiException {
        return invoke("GET",
                apiPath("instance", "services", id),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    // ========================================================================
    // Maintenance
    // ========================================================================

    public Map<String, Object> getMaintenanceStatus() throws ApiException {
        return invoke("GET",
                apiPath("instance", "maintenance", "status"),
                null, null, null,
                JSON, null,
                new TypeReference<>() {});
    }

    public void enterMaintenance() throws ApiException {
        invoke("POST",
                apiPath("instance", "maintenance", "enter"),
                null, null, null,
                JSON, null,
                null);
    }

    public void exitMaintenance() throws ApiException {
        invoke("POST",
                apiPath("instance", "maintenance", "exit"),
                null, null, null,
                JSON, null,
                null);
    }

    // ========================================================================
    // MCP servers
    // ========================================================================

    public Map<String, Object> listMcpServers(
            @jakarta.annotation.Nullable Integer page,
            @jakarta.annotation.Nullable Integer size,
            @jakarta.annotation.Nullable List<String> sort) throws ApiException {
        return invoke("GET",
                apiPath("instance", "mcp-servers"),
                null, queryParams("page", page, "size", size), csvParams("sort", sort),
                JSON, null,
                new TypeReference<>() {});
    }

}

File: python/python-sdk/kestrapy/api/instance_api.py
from typing import Any, List, Optional

from kestrapy.base_api import BaseApi
from kestrapy.models.instance_controller_api_active_service_list import InstanceControllerApiActiveServiceList
from kestrapy.models.instance_controller_api_plugin_artifact_list_plugin_artifact import InstanceControllerApiPluginArtifactListPluginArtifact
from kestrapy.models.instance_controller_api_plugin_artifact_list_plugin_resolution_result import InstanceControllerApiPluginArtifactListPluginResolutionResult
from kestrapy.models.instance_controller_api_plugin_version_details import InstanceControllerApiPluginVersionDetails
from kestrapy.models.instance_controller_api_plugin_versions import InstanceControllerApiPluginVersions
from kestrapy.models.instance_controller_api_service_instance import InstanceControllerApiServiceInstance
from kestrapy.models.maintenance_status_response import MaintenanceStatusResponse
from kestrapy.models.paged_results_instance_controller_api_plugin_artifact import PagedResultsInstanceControllerApiPluginArtifact
from kestrapy.models.paged_results_instance_controller_api_service_instance import PagedResultsInstanceControllerApiServiceInstance
from kestrapy.models.plugin_artifact import PluginArtifact
from kestrapy.models.query_filter import QueryFilter
from kestrapy.models.service_instance import ServiceInstance
from kestrapy.models.worker_credential_controller_api_worker_credential import WorkerCredentialControllerApiWorkerCredential
from kestrapy.models.worker_credential_controller_api_worker_list import WorkerCredentialControllerApiWorkerList


class InstanceApi(BaseApi):
    """Instance-wide administration (`/api/v1/instance`). Instance-owner-only.

    Covers running services, maintenance mode, versioned-plugin management,
    worker credentials and the cross-tenant MCP-server listing.
    """


    # ---- Services ----

    def search_services(
        self,
        page: Optional[int] = None,
        size: Optional[int] = None,
        sort: Optional[List[str]] = None,
        filters: Optional[List[QueryFilter]] = None,
    ) -> PagedResultsInstanceControllerApiServiceInstance:
        path = self._superadmin_path("instance", "services", "search")
        params = list(self._build_query_params(page=page, size=size).items())
        self._append_repeated_param(params, "sort", sort)
        self._append_filter_params(params, filters)
        return self._json_request("GET", path, PagedResultsInstanceControllerApiServiceInstance, params=params)

    def active_services(self) -> InstanceControllerApiActiveServiceList:
        path = self._superadmin_path("instance", "services", "active")
        return self._json_request("GET", path, InstanceControllerApiActiveServiceList)

    def service(self, id: str) -> InstanceControllerApiServiceInstance:
        path = self._superadmin_path("instance", "services", id)
        return self._json_request("GET", path, InstanceControllerApiServiceInstance)

    # ---- Maintenance ----

    def enter_maintenance(self) -> Any:
        path = self._superadmin_path("instance", "maintenance", "enter")
        return self._raw_json_request("POST", path)

    def exit_maintenance(self) -> Any:
        path = self._superadmin_path("instance", "maintenance", "exit")
        return self._raw_json_request("POST", path)

    def maintenance_status(self) -> MaintenanceStatusResponse:
        path = self._superadmin_path("instance", "maintenance", "status")
        return self._json_request("GET", path, MaintenanceStatusResponse)

    # ---- Versioned plugins ----

    def list_available_versioned_plugins(self) -> Any:
        path = self._superadmin_path("instance", "versioned-plugins", "available")
        return self._raw_json_request("GET", path)

    def list_available_versioned_plugins_for_storage(self) -> Any:
        path = self._superadmin_path("instance", "versioned-plugins", "available", "storages")
        return self._raw_json_request("GET", path)

    def list_available_versioned_plugins_for_secret_manager(self) -> Any:
        path = self._superadmin_path("instance", "versioned-plugins", "available", "secrets-managers")
        return self._raw_json_request("GET", path)

    def list_versioned_plugins(
        self,
        page: Optional[int] = None,
        size: Optional[int] = None,
        sort: Optional[List[str]] = None,
        filters: Optional[List[QueryFilter]] = None,
    ) -> PagedResultsInstanceControllerApiPluginArtifact:
        path = self._superadmin_path("instance", "versioned-plugins")
        params = list(self._build_query_params(page=page, size=size).items())
        self._append_repeated_param(params, "sort", sort)
        self._append_filter_params(params, filters)
        return self._json_request("GET", path, PagedResultsInstanceControllerApiPluginArtifact, params=params)

    def versioned_plugin_details(self, group_id: str, artifact_id: str) -> InstanceControllerApiPluginVersions:
        path = self._superadmin_path("instance", "versioned-plugins", group_id, artifact_id)
        return self._json_request("GET", path, InstanceControllerApiPluginVersions)

    def versioned_plugin_icon(self, group_id: str, artifact_id: str) -> bytes:
        path = self._superadmin_path("instance", "versioned-plugins", group_id, artifact_id, "icon.svg")
        return self._download_request("GET", path, accept="image/svg+xml")

    def versioned_plugin_details_for_version(self, group_id: str, artifact_id: str, version: str) -> InstanceControllerApiPluginVersionDetails:
        path = self._superadmin_path("instance", "versioned-plugins", group_id, artifact_id, version)
        return self._json_request("GET", path, InstanceControllerApiPluginVersionDetails)

    def versioned_plugin_release_notes(self, group_id: str, artifact_id: str) -> str:
        path = self._superadmin_path("instance", "versioned-plugins", group_id, artifact_id, "release-notes")
        return self._text_request("GET", path)

    def resolve_versioned_plugins(self, artifacts: List[PluginArtifact]) -> InstanceControllerApiPluginArtifactListPluginResolutionResult:
        path = self._superadmin_path("instance", "versioned-plugins", "resolve")
        return self._json_request("POST", path, InstanceControllerApiPluginArtifactListPluginResolutionResult, body=artifacts)

    def install_versioned_plugins(self, artifacts: List[PluginArtifact]) -> InstanceControllerApiPluginArtifactListPluginArtifact:
        path = self._superadmin_path("instance", "versioned-plugins", "install")
        return self._json_request("POST", path, InstanceControllerApiPluginArtifactListPluginArtifact, body=artifacts)

    def upload_versioned_plugin(self, file_content: Any, file_name: str = "plugin.jar") -> Optional[PluginArtifact]:
        path = self._superadmin_path("instance", "versioned-plugins", "upload")
        return self._multipart_upload("POST", path, PluginArtifact, file_content=file_content, file_name=file_name)

    def uninstall_versioned_plugins(self, artifacts: List[PluginArtifact]) -> InstanceControllerApiPluginArtifactListPluginArtifact:
        path = self._superadmin_path("instance", "versioned-plugins", "uninstall")
        return self._json_request("DELETE", path, InstanceControllerApiPluginArtifactListPluginArtifact, body=artifacts)

    # ---- MCP servers (cross-tenant) ----

    def list_all_mcp_servers(
        self,
        page: Optional[int] = None,
        size: Optional[int] = None,
        sort: Optional[List[str]] = None,
    ) -> Any:
        path = self._superadmin_path("instance", "mcp-servers")
        params = list(self._build_query_params(page=page, size=size).items())
        self._append_repeated_param(params, "sort", sort)
        return self._raw_json_request("GET", path, params=params)

    # ---- Worker credentials ----

    def list_worker_credentials(self) -> WorkerCredentialControllerApiWorkerList:
        path = self._superadmin_path("instance", "workers", "credentials")
        return self._json_request("GET", path, WorkerCredentialControllerApiWorkerList)

    def worker_credential(self, id: str) -> WorkerCredentialControllerApiWorkerCredential:
        path = self._superadmin_path("instance", "workers", "credentials", id)
        return self._json_request("GET", path, WorkerCredentialControllerApiWorkerCredential)

    def revoke_worker_credential(self, id: str) -> WorkerCredentialControllerApiWorkerCredential:
        path = self._superadmin_path("instance", "workers", "credentials", id, "revoke")
        return self._json_request("POST", path, WorkerCredentialControllerApiWorkerCredential)
